src/main/java/it/gov/pagopa/timeline/service/TimelineServiceImpl.java

```diff
@@ -36,6 +36,12 @@ public class TimelineServiceImpl implements TimelineService {
   @Autowired
   TimelineProducer timelineProducer;
 
+  private static final Set<String> ignoreCombinations = Set.of(
+      TimelineConstants.TRX_STATUS_AUTHORIZED + TimelineConstants.TRX_STATUS_REWARDED,
+      TimelineConstants.TRX_STATUS_AUTHORIZED + TimelineConstants.TRX_STATUS_AUTHORIZED,
+      TimelineConstants.TRX_STATUS_REWARDED + TimelineConstants.TRX_STATUS_REWARDED
+  );
```

Comment:
a Pair is a better structure than String concat ;) you avoid also the overhead to build at runtime the concatenated String ;)
```suggestion
  private static final Set<Pair<String>> ignoreCombinations = Set.of(
      Pair.of(TimelineConstants.TRX_STATUS_AUTHORIZED, TimelineConstants.TRX_STATUS_REWARDED),
      Pair.of(TimelineConstants.TRX_STATUS_AUTHORIZED , TimelineConstants.TRX_STATUS_AUTHORIZED),
      Pair.of(TimelineConstants.TRX_STATUS_REWARDED, TimelineConstants.TRX_STATUS_REWARDED)
  );
```